gemmapy/gemmapy_api.py

```diff
@@ -2,50 +2,105 @@
 """
 Gemma python API (https://gemma.msl.ubc.ca/rest/v2/)
 """
-
-from gemmapy import sdk
-from gemmapy import _processors as ps
-from gemmapy import _validators as vs
-from gemmapy import _subprocessors as sub
+import enum
+import json
+import logging
+import os
+import subprocess
+import warnings
+from getpass import getpass
+from io import StringIO
 from typing import Optional, List, Callable
-from pandas import DataFrame
-import pandas as pd
-import numpy as np
+
 import anndata as ad
+import numpy as np
+import pandas as pd
 from anndata import AnnData
-from io import StringIO
-import warnings
-import json
+from pandas import DataFrame
+
+from gemmapy import _processors as ps
+from gemmapy import _subprocessors as sub
+from gemmapy import _validators as vs
+from gemmapy import sdk
+
+logger = logging.getLogger(__name__)
 
+class GemmaPath(enum.Enum):
+    PROD = "prod"
+    DEV = "dev"
+    STAGING = "staging"
 
 class GemmaPy(object):
     """
     Main API class
     """
 
-    def __init__(self, auth:list|tuple=None, path="prod"):
+    def __init__(self, auth: Optional[list | tuple] = None,
```

Comment:
@oganm I changed the default for auth to be the default host in the OpenAPI specification. This matches the previous behavior of passing `prod`. However, now when passing `prod`, it will explicitly use the production path.